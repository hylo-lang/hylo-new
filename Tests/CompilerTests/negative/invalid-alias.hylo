//! no-std

struct S<T> { public memberwise init }

type X<T> = T
type Y<T> = S<T>
type Z<T> = {T, T}

let a: X<_> = ()
let b: Y<_> = S<Void>()
let c: Z<_> = (S<Void>(), S())

fun f(x: X<Void>) {}
fun g() { f(x: ()) }

fun h(x: Z<Void>) {
  f(x: x)
}
