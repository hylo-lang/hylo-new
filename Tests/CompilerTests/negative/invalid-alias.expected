negative/invalid-alias.hylo:17.8-9: error: cannot pass value of type 'Z<Void>' to parameter 'X<Void>'
  f(x: x)
       ^
